cmake/FindODBC.cmake

```diff
@@ -13,45 +13,79 @@
 # ODBC_LIBRARIES, the libraries to link against to use ODBC
 # ODBC_FOUND.  If false, you cannot build anything that requires MySQL.
 
-find_path(ODBC_INCLUDE_DIRECTORIES
-	NAMES sql.h
-	HINTS
-	/usr/include
-	/usr/include/odbc
-	/usr/include/iodbc
-	/usr/local/include
-	/usr/local/include/odbc
-	/usr/local/include/iodbc
-	/usr/local/odbc/include
-	/usr/local/iodbc/include
-	"C:/Program Files/ODBC/include"
-	"C:/Program Files/Microsoft SDKs/Windows/v7.0/include"
-	"C:/Program Files/Microsoft SDKs/Windows/v6.0a/include"
-	"C:/ODBC/include"
-	DOC "Specify the directory containing sql.h."
-)
-
-find_library(ODBC_LIBRARIES
-	NAMES iodbc odbc iodbcinst odbcinst odbc32
-	HINTS
-	/usr/lib
-	/usr/lib/odbc
-	/usr/lib/iodbc
-	/usr/local/lib
-	/usr/local/lib/odbc
-	/usr/local/lib/iodbc
-	/usr/local/odbc/lib
-	/usr/local/iodbc/lib
-	"C:/Program Files/ODBC/lib"
-	"C:/ODBC/lib/debug"
-	"C:/Program Files (x86)/Microsoft SDKs/Windows/v7.0A/Lib"
-	DOC "Specify the ODBC driver manager library here."
-)
-
-# MinGW find usually fails
-if(MINGW)
+if(WIN32)
```

Comment:
lots of windows version-specific stuff here, but is it really needed? unlike other platforms, I've never seen a problem finding ODBC headers or libraries on windows

Reply:
> I've never seen a problem finding ODBC headers or libraries on windows

Have you got a pure Win10-Installation without the SDK for Win7 and Win8? In that case the hints
```
"C:/Program Files/Microsoft SDKs/Windows/v7.0/include"
"C:/Program Files/Microsoft SDKs/Windows/v6.0a/include"
```
will not work. Win10 is organizing the SDKs in "kits" for each version, so you have to look for odbc32.lib and use the corresponding include folder.

Reply:
Maybe we should also include to set the paths `HINTS` via an environment settings ... like in MySQL ...

Reply:
When installing MySQL the corresponding environment variables for MySQL are created. When ODBC is installed, environment variables are also created. Except when using the Microsoft SDKs under Windows. There are no environment variables for ODBC. And I don't know what it looks like on OSX.